### ISSUE TITLE
动态图中更换place会导致eval状态失效

### ISSUE THREAD

**Issue by kinghuin** [PaddlePaddle/Paddle#25513]
paddle 1.8.2

with fluid.dygraph.guard(fluid.CPUPlace()):
        此处定义网络结构
        model.eval()

with fluid.dygraph.guard(fluid.CPUPlace()):
        此处调用model前向网络
        发现dropout任然生效了，说明此时model是处于train状态。
        需要在调用网络前重新model.eval()才能保证model处于eval状态

model的状态不应该和place相关？

**Comment by phlrain** [comment]
使用 with fluid.dygraph.guard， 可以理解为生成了一个新的状态，与前一个guard中的状态不一样。目前 train eval 状态是在一个guard中的状态中存储的

mode的状态目前跟guard相关，这个点确实不是非常合理，这个修复我们正在进行中

建议使用 dygraph.nn.Dropout() 不要使用layers.dropout()  即可解决你的问题

**Comment by kinghuin** [comment]
> mode的状态目前跟guard相关，这个点确实不是非常合理，这个修复我们正在进行中

预计会在哪个版本修复呢？

**Comment by phlrain** [comment]
估计会在下个小的版本中进行修复，大概1-2月。
